web/src/components/fullscreen.tsx

```diff
@@ -0,0 +1,38 @@
+import { createEffect, createSignal, onCleanup } from "solid-js"
+
+export const FullscreenButton = () => {
+	const [isActive, setIsActive] = createSignal(false)
+
+	const toggleFullscreen = () => {
+		const videoElement = document.getElementById("video")
+
+		if (!isActive()) {
+			void videoElement?.requestFullscreen()
+			setIsActive(true)
+		} else {
+			void document.exitFullscreen()
+			setIsActive(false)
+		}
+	}
+	createEffect(() => {
+		const handleFullscreenChange = () => {
+			setIsActive(!!document.fullscreenElement)
+		}
+
+		document.addEventListener("fullscreenchange", handleFullscreenChange)
+		onCleanup(() => document.removeEventListener("fullscreenchange", handleFullscreenChange))
```

Comment:
One day, it would be awesome if we could set up these events and get fullscreen from a shared store. Long-term, we could want to use this is in other areas.